File: angularOlBank/src/app/model/Message.ts
import {Forum} from './Forum';

export class Message {

  messageId!: number;
  userId!: number;
  /**
   * decision to use forum object instead of forumId
   */
  forum!:Forum;
  nbrLike!: number;
  nbrDisLike!: number;
  likingUserList!: string;
  disLikingUserList!: string;
  dateTime!: Date;
  messageTxt!: string;

constructor(userId: number,forum: Forum, messageTxt: string) {
  this.userId=userId;
  this.forum=forum;
  this.nbrLike=0;
  this.nbrDisLike=0;
  this.likingUserList='';
  this.disLikingUserList='';
  this.dateTime= new Date();
  this.messageTxt=messageTxt;
}
}

File: springOlbank/src/main/java/com/spark3/olbank/forum/model/Message.java
package com.spark3.olbank.forum.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "message")
@Getter
@Setter
@AllArgsConstructor
@ToString

public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long messageId;
    private long userId;
    private int nbrLike;
    @Column(nullable = true)
    private int nbrDisLike;
    private String likingUserList;
    @Column(nullable = true)
    private String disLikingUserList;
    private Date dateTime;
    private String messageTxt;
    @ManyToOne
    @JoinColumn(name="forumId",referencedColumnName="forumId")
    @JsonIgnoreProperties({"listeMessages","topic", "forumDate"})
    private Forum forum;
    //@JsonIgnore
    public Message() {
    }
    public Message(String s) {
        this.messageTxt = s;
    }

}
